app/controllers/course/survey/surveys_controller.rb

```diff
@@ -84,7 +84,8 @@ def survey_params
       :title, :description, :base_exp, :time_bonus_exp, :start_at, :bonus_end_at, :end_at,
       :published, :allow_response_after_end, :allow_modify_after_submit
     ]
-    fields << :anonymous if !@survey.anonymous || !@survey.has_student_response?
+    fields << :anonymous if action_name == 'create' || !@survey.anonymous ||
```

Comment:
I don't think it is very critical. It's pseudo-anonymous anyway. Actually, what you mentioned was one of the reasons I prefer not to add the validation - I imagine there might be a case where a staff member will ask for help to change the status because s/he changed her mind about the status after a couple of students have submitted. In that case, admins should be able to help without too much hassle.

You are right about the other services. It is a potential risk.